pkg/controllermanager/controller/shootstate/add_test.go

```diff
@@ -0,0 +1,195 @@
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/shootstate"
+)
+
+var _ = Describe("Add", func() {
+	var (
+		ctx        context.Context
+		c          client.Client
+		reconciler *shootstate.Reconciler
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		c = fakeclient.NewClientBuilder().WithScheme(kubernetes.GardenScheme).Build()
+		reconciler = &shootstate.Reconciler{
+			Client: c,
+		}
+	})
+
+	Describe("#MapShootToShootState", func() {
+		var (
+			ctx context.Context
+			log logr.Logger
+
+			shootName      = "shoot-1"
+			shootNamespace = "ns-1"
+		)
+
+		BeforeEach(func() {
+			ctx = context.Background()
+			log = logr.Discard()
+		})
+
+		It("should return reconciliation request matching the shoot name", func() {
+			shoot := &gardencorev1beta1.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      shootName,
+					Namespace: shootNamespace,
+				},
+			}
+
+			reconciliationRequest := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      shootName,
+					Namespace: shootNamespace,
+				},
+			}
+
+			shootState := &gardencorev1beta1.ShootState{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      shootName,
+					Namespace: shootNamespace,
+				},
+			}
+			Expect(c.Create(ctx, shootState)).To(Succeed())
+
+			requests := reconciler.MapShootToShootState(log)(ctx, shoot)
+			Expect(requests).To(ConsistOf(reconciliationRequest))
+		})
+
+		It("should return nil if ShootState is not present", func() {
+			shoot := &gardencorev1beta1.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      shootName,
+					Namespace: shootNamespace,
+				},
+			}
+			requests := reconciler.MapShootToShootState(log)(ctx, shoot)
+			Expect(requests).To(BeEmpty())
+		})
+
+		It("should return nil if argument is not a gardencorev1beta1.Shoot", func() {
+			deploymentName := "dep-1"
+			deploymentNamespace := "ns-1"
+			deployment := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deploymentName,
+					Namespace: deploymentNamespace,
+				},
+			}
+
+			requests := reconciler.MapShootToShootState(log)(ctx, &deployment)
+			Expect(requests).To(BeNil())
+		})
+	})
+
+	Describe("ShootPredicate", func() {
+		var p predicate.Predicate
+
+		BeforeEach(func() {
+			p = reconciler.ShootPredicates()
+		})
+
+		Describe("#Update", func() {
+			var (
+				shoot    *gardencorev1beta1.Shoot
+				shootOld *gardencorev1beta1.Shoot
+
+				shootState *gardencorev1beta1.ShootState
+			)
+
+			BeforeEach(func() {
+				shoot = &gardencorev1beta1.Shoot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "shoot-",
+						Namespace: "ns-",
+					},
+					Status: gardencorev1beta1.ShootStatus{},
```

Comment:
```suggestion
```
Unnecessary as the status is already defaulted to an empty structure (it's not a pointer)

Reply:
Thanks for pointing this redundant assignment, removed it.